### PR TITLE
Changed exp calculation and added an option for exp scaling on multiply exp; fixed #53

### LINKED ISSUE #53: Zombie'ish entities glitching through walls on kill.
**Brief Description:**
pig zombies, zombie villagers and regular zombies glitch through walls when killed while stacked

**Instructions:**
have a stack of the above mentioned and kill them close to a wall

**How many times did you recreate this?:**
at least 10 times

**Result:**
the amount of mobs die as defined in the stackmob plugin and sometimes the remaining stack glitches through a wall

**Expected Result:**
mobs to stay in place like any other mob does (this does infact not happen to any of the other mobs)

**Evidence:**
Before:
https://i.imgur.com/m1QDvFy.png
After:
https://i.imgur.com/CVAy7cx.png

### LINKED ISSUE #61: Not receiving all the exp from kill-all/kill-step.
Any help with this would be greatly appreciated.

My configuration file can be seen below:
https://pastebin.com/raw/ssKaBjWM

kill-all/kill-step is enabled with specific whitelists, as well the following option is true:
_multiply-exp-enabled: true_

We have recently received the following bug report:

**Brief Description:**
killing >1 mobs from a stack only yields exp for the first and a fraction for every other mob killed

**Instructions:**
have a stack of mobs (for the ease of testing i recommend blaze as they yield exactly 10exp per mob)
kill >1
see how you only get 10 for the first mob a fraction of exp for every additional mob killed

**How many times did you recreate this?:**
couple times, one time i took the following screenshots

**Result:**
gained 12 exp after killing 4 blazes

**Expected Result:**
gain 40 exp as supposed to (referring to the test i made the screenshots with)
or whatever scaling the plugin config had before the update of the plugin (i know that kill-step in the config was messed up as well after updating)

**Evidence:**

i made sure the spawner was actually not spawning to avoid having faulty results due to restacking.

**amount in stack before:**
https://i.imgur.com/G4T0tBH.png

**amount of exp before:**
https://i.imgur.com/RBLow5r.png

**amount in stack after:**
https://i.imgur.com/8Dyr2Ox.png

**amount of exp after:**
https://i.imgur.com/RP4xbqq.png

@gluupor if you have anything to add please let us know :)

### PR DESCRIPTION
**Pull request by Gluupor**
Should fix #61 adds some functionality

### CODE REVIEW

**Comment by andrewkm** [comment]
@Gluupor have you tested this locally?

This looks like it will error for anyone who doesn't update the config as it has no default as well.

**Comment by Gluupor** [comment]
made an edit to provide a default thx @andrewkm and @jamiesinn for pointing that out

**Comment by Gluupor** [comment]
https://github.com/Nathat23/StackMob-2/pull/62/commits/97f8d0ae26af8a1f2a6dba6482e4e9288e4d6f5b
i tested this for a good 20 mins locally and didnt have any zombie'ish entities glitching through a wall.

the issue remains with bigger entities like horses, donkeys and such (yes there actually is a problem with them as well even before the change i made and tested). dealing with that would require more intenisve checking and figuring out a decent location to spawn the new entitiy in. then again this would be impossible when people drop those kind of entities in a 1x1 block space and then proceed to kill them.

File: src/main/java/uk/antiperson/stackmob/listeners/entity/DeathEvent.java [hunk 1]
```diff
@@ -80,8 +82,8 @@ private void multiplication(LivingEntity dead, List<ItemStack> drops, int subtra
             }
         }
         if(sm.config.getCustomConfig().getBoolean("multiply-exp-enabled")){
-            int newExperience = (int) Math.round(originalExperience * (subtractAmount - 1) * (0.75 + ThreadLocalRandom.current().nextDouble(0.5)));
-            ((ExperienceOrb) dead.getWorld().spawnEntity(dead.getLocation(), EntityType.EXPERIENCE_ORB)).setExperience(newExperience);
+        	double newExperience = subtractAmount * (originalExperience * sm.config.getCustomConfig().getDouble("multiply-exp-scaling", 1.0));
+            ((ExperienceOrb) dead.getWorld().spawnEntity(dead.getLocation(), EntityType.EXPERIENCE_ORB)).setExperience((int)newExperience);
```
Comment (Nathat23) [inline src/main/java/uk/antiperson/stackmob/listeners/entity/DeathEvent.java]: Casting a double to an int without rounding is a bad idea.

File: src/main/java/uk/antiperson/stackmob/listeners/entity/DeathEvent.java [hunk 2]
```diff
@@ -10,9 +10,11 @@
 import org.bukkit.event.entity.EntityDeathEvent;
 import org.bukkit.inventory.ItemStack;
 import org.bukkit.metadata.FixedMetadataValue;
+
 import uk.antiperson.stackmob.StackMob;
 import uk.antiperson.stackmob.tools.extras.GlobalValues;
 
+import java.io.Console;
```
Comment (Nathat23) [inline src/main/java/uk/antiperson/stackmob/listeners/entity/DeathEvent.java]: Unused import?

**Pull request merged**